### ISSUE TITLE
Network: how to define arrays nodes and edges of other way (equivalent) ?

### ISSUE THREAD

**Issue by robertopg15** [almende/vis#1237]
hi
my name is Roberto ... from chile
in parentheses ... i do not speak english ... but i'm using an online translator ... so if something is not written well ... I apologize ...
ok

my question is ... about the array "nodes" (and "edges") in javascirpt to create an DataSet ... specifically, what should i consider as an element of the array nodes ?

to be understood ...

I want to create a dynamic network, then to create the structure of the network vis in php ... entry the nodes names from an input html ...  later i send the data of the database to another php file which contains JavaScript code, where is the DataSet and everything the rest of code to generate the network ... and well ... when creating an array, not really know that should I put inside of the array

______________
based on
var nodes = new vis.DataSet([
        {id: 1, label: 'Node 1'},
        {id: 2, label: 'Node 2'}
    ]);

which works ok
______________

To develop what I said above, first I need to understand which is the structure vis and how it operates...

so ... if i refer only to javascript, how i could define the array ... 
[{id: 1, label: 'Node 1'},{id: 2, label: 'Node 2'}]
... equivalently?, that is, define the same array but of other way (equivalent) ... for then pass it to the DataSet 

I tried the following and it did not work correctly (It shows the two nodes but not the text and without the edges)

<script type="text/javascript">
    
    //for nodes
    var aux1 = new Array(
                        new Array("id: 1", "label: 'Node 1'"), 
                        new Array("id: 2", "label: 'Node 2'")
                        );
    var nodes = new vis.DataSet(aux1);

    //for edges	    
    var aux2 = new Array(
           		new Array("from: 1", "to: 2") 
                        );
    var edges = new vis.DataSet(aux2);
    // create a network
    var container = document.getElementById('mynetwork');

    // provide the data in the vis format
    var data = {
        nodes: nodes,
        edges: edges
    };
    var options = {};

    // initialize your network!
    var network = new vis.Network(container, data, options);
</script>

What am I doing wrong?

I tried also combinations

  //for nodes
    var aux1 = new Array(
                        new Array("1", "'Node 1'"), 
                        new Array("2", "'Node 2'")
                        );
    var nodes = new vis.DataSet(aux1);

and 

    var aux1 = new Array(
                        new Array(1, Node 1), 
                        new Array(2, Node 2)
			);
    var nodes = new vis.DataSet(aux1);

and 

var aux1 = new Array(
                        new Array("id: 1", "label: Node 1"), 
                        new Array("id: 2", "label: Node 2")
			);
    var nodes = new vis.DataSet(aux1);

and

    var aux1 = new Array(
                        new Array("{id: 1}", "{label: Node 1}"), 
                        new Array("{id: 2}", "{label: Node 2}")
			);
    var nodes = new vis.DataSet(aux1);

and 
	
var aux1 = new Array(
                     new Array("[{id: 1}]", "[{label: Node 1}]"), 
                     new Array("[{id: 2}]", "[{label: Node 2}]")
               	);
    var nodes = new vis.DataSet(aux1);

...

so i need know how to solve the problem, that is, how is defined an array "nodes" and "edges" correctly ?, What are its elements? (I should assume that both arrays are worked in the same way)

in other words ...  understand how the library vis works internally 

can someone help me plese ?

I thank you very much

**Comment by AlexDM0** [comment]
Hi,

This has nothing to do with the internals of the library. I think what you need is:

var dataArray = [{id:1,label:"foo"},{id:2,label:"bar"}];

then

var nodes = new vis.DataSet(dataArray);

This is basic JavaScript, I'd recommend you go back to the basics and go from there. Look at JavaScript arrays and objects and how to define them. 

Good luck

**Comment by AlexDM0** [comment]
Keep in mind: 

```
var a = {b:34};
```

is NOT an array, it is a javascript object.

Regards
